deploy/modules/gcp/tidb-operator/main.tf

```diff
@@ -0,0 +1,144 @@
+    }
+  }
+
+  ip_allocation_policy {
+    use_ip_aliases = true
+  }
+
+  // See https://kubernetes.io/docs/setup/best-practices/cluster-large/#size-of-master-and-master-components for why initial_node_count is 5. The master node should accommodate up to 100 nodes like this
+  remove_default_node_pool = true
+  initial_node_count       = 5
+
+  min_master_version = var.gke_version
+
+  lifecycle {
+    ignore_changes = [master_auth] // see above linked issue
+  }
+
+  maintenance_policy {
+    daily_maintenance_window {
+      start_time = "01:00"
+    }
+  }
+}
+
+resource "null_resource" "get-credentials" {
+  depends_on = [google_container_cluster.cluster]
+  provisioner "local-exec" {
+    command = "gcloud container clusters get-credentials ${google_container_cluster.cluster.name} --region ${var.gcp_region}"
+
+    environment = {
+      KUBECONFIG = var.kubeconfig_path
+    }
+  }
+
+  provisioner "local-exec" {
+    when = destroy
+
+    command = <<EOS
+kubectl get pvc -n tidb -o jsonpath='{.items[*].spec.volumeName}'|fmt -1 | xargs -I {} kubectl patch pv {} -p '{"spec":{"persistentVolumeReclaimPolicy":"Delete"}}'
+EOS
+
+
+    environment = {
+      KUBECONFIG = var.kubeconfig_path
+    }
+  }
+}
+
+provider "helm" {
+  alias    = "initial"
+  insecure = true
+  # service_account = "tiller"
+  install_tiller = false # currently this doesn't work, so we install tiller in the local-exec provisioner. See https://github.com/terraform-providers/terraform-provider-helm/issues/148
+  kubernetes {
+    config_path = var.kubeconfig_path
+  }
+}
+
+resource "null_resource" "setup-env" {
+  depends_on = [
+    google_container_cluster.cluster,
+    null_resource.get-credentials,
+    var.tidb_operator_registry,
+    var.tidb_operator_version,
+  ]
+
+  provisioner "local-exec" {
+    working_dir = path.cwd
+    interpreter = ["bash", "-c"]
+
+    command = <<EOS
+set -euo pipefail
+
+if ! kubectl get clusterrolebinding cluster-admin-binding 2>/dev/null; then
+  kubectl create clusterrolebinding cluster-admin-binding --clusterrole cluster-admin --user $(gcloud config get-value account)
+fi
+
+if ! kubectl get serviceaccount -n kube-system tiller 2>/dev/null ; then
+  kubectl create serviceaccount --namespace kube-system tiller
+fi
+
+kubectl apply -f https://raw.githubusercontent.com/pingcap/tidb-operator/${var.tidb_operator_version}/manifests/crd.yaml
+kubectl apply -f https://raw.githubusercontent.com/pingcap/tidb-operator/${var.tidb_operator_version}/manifests/tiller-rbac.yaml
+kubectl apply -k manifests/local-ssd
+kubectl apply -f manifests/gke/persistent-disk.yaml
+
+helm init --service-account tiller --upgrade --wait
+until helm ls; do
+  echo "Wait until tiller is ready"
+  sleep 5
+done
+EOS
+
+
+    environment = {
+      KUBECONFIG = var.kubeconfig_path
+    }
+  }
+}
+
+data "helm_repository" "pingcap" {
+  provider = "helm.initial"
+  depends_on = ["null_resource.setup-env"]
+  name = "pingcap"
+  url = "http://charts.pingcap.org/"
```

Comment:
prefer `https` (the other terraform scripts is using `http` because `https` is absent at the writing time)